File: Lessons/05_PrefixSums_PassingCars.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
    # write your code in Ruby 2.2
    ans =  0
    eastCnt = 0 
    a.length.times do |i|
      if a[i] == 0 
        eastCnt += 1 
      else 
        ans += eastCnt
      end
    end
    return ans <= 1000000000 ? ans : -1
end
  

File: Python/05_PrefixSums_PassingCars.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    cnt = 0 
    ans = 0
    for a in A:
        if a == 0: cnt += 1
        else: ans += cnt   
    
    return ans if ans <= 10 ** 9 else -1
# end of func
            
